lyndapham/step/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,18 +50,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     List<Vote> votes = new ArrayList<>();
+    boolean hidden = false;
+
     for (Entity entity : results.asIterable()) {
       String vote = (String) entity.getProperty("vote");
       String comment = (String) entity.getProperty("comment");
 
-      Vote entry = new Vote(vote, comment);
+      double score = (double) entity.getProperty("sentimentScore");
+      Vote entry = new Vote(vote, comment, score);
+      if (score < -0.6) {
+        hidden = true;
+        continue;
+      }
       votes.add(entry);
     }
 
     Gson gson = new Gson();
+    if (!votes.isEmpty()) {
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(votes));
+    }
 
-    response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(votes));
+    if (hidden) {
+      response.setContentType("text/html;");
```

Comment:
if there are votes but some of them are hidden is the response type supposed to be "text/html". As I understand it there can only be one content-type per response. It would probably make sense to have a VoteResponse type which wraps the List of Votes and has an optional field for error messages which you can then gson.

Reply:
Created a VoteResponse class that has the List and error String as fields!